addons/product/pricelist.py

```diff
@@ -344,95 +256,20 @@ def price_rule_get(self, cr, uid, ids, prod_id, qty, partner=None, context=None)
-    }
-
-    def _check_date(self, cursor, user, ids, context=None):
-        for pricelist_version in self.browse(cursor, user, ids, context=context):
-            if not pricelist_version.active:
-                continue
-            where = []
-            if pricelist_version.date_start:
-                where.append("((date_end>='%s') or (date_end is null))" % (pricelist_version.date_start,))
-            if pricelist_version.date_end:
-                where.append("((date_start<='%s') or (date_start is null))" % (pricelist_version.date_end,))
-
-            cursor.execute('SELECT id ' \
-                    'FROM product_pricelist_version ' \
-                    'WHERE '+' and '.join(where) + (where and ' and ' or '')+
-                        'pricelist_id = %s ' \
-                        'AND active ' \
-                        'AND id <> %s', (
-                            pricelist_version.pricelist_id.id,
-                            pricelist_version.id))
-            if cursor.fetchall():
-                return False
-        return True
-
-    _constraints = [
-        (_check_date, 'You cannot have 2 pricelist versions that overlap!',
-            ['date_start', 'date_end'])
-    ]
-
-    def copy(self, cr, uid, id, default=None, context=None):
-        # set active False to prevent overlapping active pricelist
-        # versions
-        if not default:
-            default = {}
-        default['active'] = False
-        return super(product_pricelist_version, self).copy(cr, uid, id, default, context=context)
-
 class product_pricelist_item(osv.osv):
-    def _price_field_get(self, cr, uid, context=None):
-        pt = self.pool.get('product.price.type')
-        ids = pt.search(cr, uid, [], context=context)
-        result = []
-        for line in pt.browse(cr, uid, ids, context=context):
-            result.append((line.id, line.name))
-
-        result.append((-1, _('Other Pricelist')))
-        return result
 
-# Added default function to fetch the Price type Based on Pricelist type.
-    def _get_default_base(self, cr, uid, fields, context=None):
-        product_price_type_obj = self.pool.get('product.price.type')
-        if fields.get('type') == 'purchase':
-            product_price_type_ids = product_price_type_obj.search(cr, uid, [('field', '=', 'standard_price')], context=context)
-        elif fields.get('type') == 'sale':
-            product_price_type_ids = product_price_type_obj.search(cr, uid, [('field','=','list_price')], context=context)
-        else:
-            return -1
-        if not product_price_type_ids:
-            return False
-        else:
-            pricetype = product_price_type_obj.browse(cr, uid, product_price_type_ids, context=context)[0]
-            return pricetype.id
+    def onchange_product(self, cr, uid, ids, categ_id, product_tmpl_id, context=None):
+        domain = {}
+        domain['product_tmpl_id'] = [('categ_id', '=', categ_id)] if categ_id else False
```

Comment:
this method should be change. cannot we just check like:
if categ_id:
    product_tmpl_domain =  [('categ_id', '=', categ_id)]
   produt_id_domain = [('product_tmpl_id', '=', product_tmpl_id)]
then return {'domain': {'product_id': ... , 'product_tmpl_id': ...}}